cookbook/docs/index.md

```diff
@@ -0,0 +1,437 @@
+
+import flytekit.extras.sklearn
+from flytekit import task, workflow
+
+
+@task
+def get_data() -> pd.DataFrame:
+    """Get the wine dataset."""
+    return load_wine(as_frame=True).frame
+
+@task
+def process_data(data: pd.DataFrame) -> pd.DataFrame:
+    """Simplify the task from a 3-class to a binary classification problem."""
+    return data.assign(target=lambda x: x["target"].where(x["target"] == 0, 1))
+
+@task
+def train_model(data: pd.DataFrame, hyperparameters: dict) -> LogisticRegression:
+    """Train a model on the wine dataset."""
+    features = data.drop("target", axis="columns")
+    target = data["target"]
+    return LogisticRegression(**hyperparameters).fit(features, target)
+```
+
+As we can see in the code snippet above, we defined three tasks as Python
+functions: `get_data`, `process_data`, and `train_model`.
+
+In Flyte, **tasks** are the most basic unit of compute and serve as the building
+blocks 🧱 for more complex applications. A task is a function that takes some
+inputs and produces an output. We can use these tasks to define a simple model
+training workflow:
+
+```{code-cell} python
+@workflow
+def training_workflow(hyperparameters: dict) -> LogisticRegression:
+    """Put all of the steps together into a single workflow."""
+    data = get_data()
+    processed_data = process_data(data=data)
+    return train_model(
+        data=processed_data,
+        hyperparameters=hyperparameters,
+    )
+```
+
+```{note}
+A task can also be an isolated piece of compute that takes no inputs and
+produces no output, but for the most part to do something useful a task
+is typically written with inputs and outputs.
+```
+
+A **workflow** is also defined as a Python function, and it specifies the flow
+of data between tasks and, more generally, the dependencies between tasks 🔀.
+
+::::{dropdown} {fa}`info-circle` The code above looks like Python, but what do `@task` and `@workflow` do exactly?
+:title: text-muted
+:animate: fade-in-slide-down
+
+Flyte `@task` and `@workflow` decorators are designed to work seamlessly with
+your code-base, provided that the *decorated function is at the top-level scope
+of the module*.
+
+This means that you can invoke tasks and workflows as regular Python methods and
+even import and use them in other Python modules or scripts.
+
+:::{note}
+A {func}`task <flytekit.task>` is a pure Python function, while a {func}`workflow <flytekit.workflow>`
+is actually a [DSL](https://en.wikipedia.org/wiki/Domain-specific_language) that
+only supports a subset of Python's semantics. Some key things to learn here are:
+
+- In workflows, you can't use non-deterministic operations like `rand.random`,
+  `time.now()`, etc.
+- Within workflows, the outputs of tasks are promises under the hood, so you
+  can't access and operate on them like typical Python function outputs.
+  *You can only pass promises into other tasks/workflows.*
+- Tasks can only be invoked with keyword arguments, not positional arguments.
```

Comment:
This is covered in Flyte fundamentals as well. Can we just redirect the users to the concerning pages instead of repeating the content?